ingredient/ingredient-app-insights/src/functions.ts

```diff
@@ -1,12 +1,41 @@
-import {BaseUtility, IngredientManager} from '@azbake/core'
+import { BaseUtility, IngredientManager } from '@azbake/core'
+import { ApplicationInsightsManagementClient } from '@azure/arm-appinsights'
+import { utils } from 'mocha';
 
 export class AppInsightsUtils extends BaseUtility {
-    
-    public create_resource_name(): string {
+
+    public get_resource_name(shortName: string): string {
         let util = IngredientManager.getIngredientFunction("coreutils", this.context)
 
-        const name = util.create_resource_name("ai", null, true);
-        return name;
+        //App Insights telemetry will be centralized.  Therefore don't appenda  region code.
+        const appInsightsName = util.create_resource_name("ai", shortName, false);
+        return appInsightsName;
     }
-}
 
+    public async get_instrumentation_key(shortName: string, rgShortName: string | null = null): Promise<string> {
+
+        let util = IngredientManager.getIngredientFunction("coreutils", this.context)
+        
+        //App Insights telemetry will be centralized.  Therefore don't appenda  region code.
+        const aiName = util.create_resource_name("ai", shortName, false);
+        
+        let rgName: string;
+
+        let override = this.context.Config.rgOverride
```

Comment:
corutils has a function for getting the resource group name, which does the override check or uses the package rg. Should be using that.

Reply:
The resource_group function in coreutils currently only supports region specific resource groups.  App Insights is centralized so I put the logic in its ingredient.  I'll parameterize the coreutils function to support both region and non-region specific ingredients.

Reply:
Also parameterizing short name since it will be pulling the resource group of the app insights resource as opposed to the resource group of an ingredient being deployed